ui/app/js/controllers/required_signoffs_delete.js

```diff
@@ -0,0 +1,156 @@
+    $scope.errors = {};
+
+    var service = null;
+    if ($scope.mode === "channel") {
+      service = ProductRequiredSignoffs;
+    }
+    else if ($scope.mode === "permissions") {
+      service = PermissionsRequiredSignoffs;
+    }
+    else {
+      $scope.errors["exception"] = "Couldn't detect mode";
+      return;
+    }
+
+    CSRF.getToken()
+    .then(function(csrf_token) {
+      $scope.saving = true;
+      var promises = [];
+
+      var successCallback = function(data, deferred) {
+        return function(response) {
+          var namespace = null;
+          if ($scope.mode === "channel") {
+            namespace = required_signoffs[$scope.product]["channels"][$scope.channel];
+          }
+          else {
+            namespace = required_signoffs[$scope.product]["permissions"];
+          }
+
+          // Required Signoff wasn't pending, so a deletion will be a new Scheduled Change
+          if (namespace[data["role"]]["sc"] === null) {
+            namespace[data["role"]]["sc"] = {
+              "required_signoffs": {},
+              "signoffs_required": 0,
+              "sc_id": response["sc_id"],
+              "scheduled_by": current_user,
+              "sc_data_version": 1,
+              "signoffs": {},
+              "change_type": "delete",
+            };
+          }
+          // Required Signoff was previously pending, so it was directly removed
+          else {
+            // If it was *only* a scheduled change, we delete the whole thing
+            if (namespace[data["role"]]["data_version"] === null) {
+              delete namespace[data["role"]];
+            }
+            // If it was a scheduled change to an existing Required Signoff, just remove
+            // the Scheduled Change portion
+            else {
+              namespace[data["role"]]["sc"] = null;
+            }
+          }
+
+          deferred.resolve();
+        };
+      };
+      var errorCallback = function(data, deferred) {
+        return function(response, status) {
+          if (typeof response === "object") {
+            $scope.errors = response;
+          }
+          else if (typeof response === "string"){
+            $scope.errors["exception"] = response;
+          }
+          else {
+            sweetAlert("Unknown error occurred");
+          }
+          deferred.resolve();
+        };
+      };
+
+      Object.keys($scope.to_delete).forEach(function(role_name) {
+        var deferred = $q.defer();
+        promises.push(deferred.promise);
+        var data = {"product": $scope.product, "role": role_name, "data_version": $scope.to_delete[role_name]["data_version"],
+                    "csrf_token": csrf_token};
+
+        if ($scope.mode === "channel") {
+          data["channel"] = $scope.channel;
+        }
+        // Not a Scheduled Change, we'll have to create one to delete it
+        if ($scope.to_delete[role_name]["sc"] === null) {
+          data["when"] = new Date().getTime() + 5000;
```

Comment:
~~I'm not sure to understand why we need to add 5s in this case. What does this time compensate?~~

Edit: Found the answer there https://github.com/mozilla/balrog/pull/253/files#diff-e729d8dd6ca1733c654cfbef188401beR285

I agree with this comment https://github.com/mozilla/balrog/pull/253/files#diff-45c43c60c890f3114299e2a5c7cd5e5dR6, an object that represents the payload would help in having this logic at one single place.